backend/internal/controllers/img.go

```diff
@@ -0,0 +1,74 @@
+package controllers
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/TritonSE/words-alive/internal/database"
+	"github.com/TritonSE/words-alive/internal/utils"
+
+	"github.com/go-chi/chi"
+)
+
+type ImgController struct {
+	Image database.ImgDatabase
+}
+
+var allowedTypes = map[string]bool{
+	"image/png":  true,
+	"image/jpeg": true,
+}
+
+// gets an image from the database (/image/{id})
+func (c *ImgController) GetImage(rw http.ResponseWriter, req *http.Request) {
+	var id string = chi.URLParam(req, "id")
+
+	img, ctype, err := c.Image.GetImage(req.Context(), id)
+
+	if err != nil {
+		writeResponse(rw, http.StatusNotFound, "Image with requested id not found")
+		return
+	}
```

Comment:
This is potentially misleading - if the database is down it will return 404s on images that exist, instead of a more-appropriate 500.

Instead can you maybe return `img=nil, ctype="", err=nil` for the case where the image doesn't exist, and reserve `err!=nil` for database errors?